clojure-mode.el

```diff
@@ -352,6 +352,21 @@ Called by `imenu--generic-function'."
               (set-match-data (list def-beg def-end)))))
         (goto-char start)))))
 
+(eval-and-compile
+  (defconst clojure-sym-rest-chars "^][\";\'@\\^`~\(\)\{\}\\"
+    "A black list of chars a clojure symbol must not contain. See
+definiton of 'macros' https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/LispReader.java#L88")
+  (defconst clojure-sym-1st-chars (concat clojure-sym-rest-chars "0-9")
+    "A black list of chars a clojure symbol must not start with. See
+getMacro invocation:
+https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/LispReader.java#L260
```

Comment:
Well, basically I took a look at the for-loop https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/LispReader.java#L232 consuming the input string in order to see which chars can be a part of a clojure symbol and which chars delimit string boundaries, lists, vectors, special forms etc. From this point of view the line 232 (together with line 254) is the key point of the for-loop.